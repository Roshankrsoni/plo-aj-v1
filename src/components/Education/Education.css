.subsection{
    margin-top:2.75rem;
}
.subsection-wrappers + .subsection-wrappers{
    margin-top:3rem;
}
.education .subsection-data{
    margin-top:1rem;
    margin-bottom:2rem;
}
.education-bars{
    margin-top:2rem;
    margin-bottom: 2rem;
}
.know-more-button{
    display: flex;
    justify-content: flex-end;
}
.know-more-button > img {
    width: 2rem;
}
