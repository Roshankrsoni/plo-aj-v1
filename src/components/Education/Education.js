import React, {useState, useEffect} from "react";
// import KnowMoreButton from "../../assets/arrow-icon.svg";
import BarGraph from "../BarGraph/BarGraph";
import "./Education.css";
import leftPaneData from '../LeftPane/leftPaneData'

function Education(props) {
  let [sectionStep, setSectionStep] = useState();
  useEffect(()=>{
      let key = 0;
      leftPaneData.forEach(ele=>{
          if(ele.label === 'Education'){
              key = ele.key;
          }
      })
      setSectionStep(parseInt(key));
  },[])

  return (
    <div
      className={
        "show-on-scroll col-md-7 page-1 text-left section education " +
        props.class
      }
    >
      <div className="section-title grey4 h2 bold">Education</div>

      <div className="subsection">
        {props.dbData.education.map((ele,key) => {
          return (
            <div className="subsection-wrappers" key={key}>
              <div className="subsection-title uppercase body-text grey3 letterspacing-1 mt-2">
                {ele.type}
              </div>
              <div className="subsection-data">
                <span className="h3 grey1 bold">{ele.institute}</span>
                <span className="h3 grey3 light">, {ele.place}</span>
                <div className="education-bars">
                  <BarGraph
                    value={ele.percentage}
                    currentStep={props.currentStep}
                    sectionStep={sectionStep}
                  />
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
}

export default Education;
