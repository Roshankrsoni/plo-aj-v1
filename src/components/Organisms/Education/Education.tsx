import React from "react";
import BarGraph from "../../Atoms/BarGraph/BarGraph";
import "./Education.css";
import { useGlobalContext } from "../../../context/GlobalContext";
import SectionVisibilityHOC from "../SectionInViewWrapper/SectionVisibilityHOC";

function Education() {
  const { profileData } = useGlobalContext();
  const { education } = profileData;
  return (
    <>
      <div className="section-title h2 bold">Education</div>
      <div className="subsection">
        {education.map((educationDetail, key) => {
          return (
            <div className="subsection-wrappers" key={key}>
              <div className="subsection-title uppercase body-text grey3 letterspacing-1 mt-2">
                {educationDetail.type}
              </div>
              <div className="subsection-data">
                <span className="h3 grey1 bold">
                  {educationDetail.institute}
                </span>
                <span className="h3 grey3 light">
                  , {educationDetail.place}
                </span>
                <div className="education-bars">
                  <BarGraph
                    value={educationDetail.percentage}
                    currentStep={1}
                    sectionStep={1}
                  />
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </>
  );
}
export default SectionVisibilityHOC(Education, "education");
