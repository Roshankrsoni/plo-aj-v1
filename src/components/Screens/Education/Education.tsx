import React from "react";
import { useGlobalContext } from "../../../context/GlobalContext";
import constants from "../../../helpers/constants";
import T from "../../../translations/en_IN";
import SectionWrapper from "../../Organisms/SectionWrapper/SectionWrapper";
import "./Education.scss";
function Education() {
  const { profileData } = useGlobalContext();
  const { education } = profileData;

  return (
    <>
      <div className="section-title h2 bold">{T.EDUCATION}</div>
      <div className="subsection">
        {education.map((educationDetail, key) => {
          return (
            <div className="subsection-wrappers" key={String(key)}>
              <div className="subsection-title uppercase body-text grey5 letterspacing-1 mt-2">
                {educationDetail.type}
              </div>
              <div className="subsection-data">
                <span className="h3 grey1 bold">
                  {educationDetail.institute}
                </span>
                <span className="h3 grey5 light">
                  , {educationDetail.place}
                </span>
              </div>
            </div>
          );
        })}
      </div>
    </>
  );
}
export default SectionWrapper(Education, constants.classNames.EDUCATION);
