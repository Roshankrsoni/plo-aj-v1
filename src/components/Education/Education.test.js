import React from 'react';
import { shallow } from 'enzyme';
import Education from './Education';

import dummyData from '../TestData/testData.json'

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Education dbData={dummyData} />);
        expect(component).toMatchSnapshot();
    });
});