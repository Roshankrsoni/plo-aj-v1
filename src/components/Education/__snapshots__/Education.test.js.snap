// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="show-on-scroll col-md-7 page-1 text-left section education undefined"
>
  <div
    className="section-title grey4 h2 bold"
  >
    Education
  </div>
  <div
    className="subsection"
  >
    <div
      className="subsection-wrappers"
      key="0"
    >
      <div
        className="subsection-title uppercase body-text grey3 letterspacing-1 mt-2"
      >
        Bachelors in Computer Science & Engineering
      </div>
      <div
        className="subsection-data"
      >
        <span
          className="h3 grey1 bold"
        >
          MIT Academy of Engineering
        </span>
        <span
          className="h3 grey3 light"
        >
          , 
          Alandi
        </span>
        <div
          className="education-bars"
        >
          <BarGraph
            value="64"
          />
        </div>
      </div>
    </div>
    <div
      className="subsection-wrappers"
      key="1"
    >
      <div
        className="subsection-title uppercase body-text grey3 letterspacing-1 mt-2"
      >
        higher secondary
      </div>
      <div
        className="subsection-data"
      >
        <span
          className="h3 grey1 bold"
        >
          Army Public School
        </span>
        <span
          className="h3 grey3 light"
        >
          , 
          Kirkee
        </span>
        <div
          className="education-bars"
        >
          <BarGraph
            value="77"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
